libgnucash/core-utils/binreloc.c

```diff
@@ -106,22 +106,23 @@ _br_find_exe (Gnc_GbrInitError *error)
     char *line, *result;
     size_t buf_size = PATH_MAX + 1;
     FILE *f;
-    uint32_t size2;
 
 #ifdef MAC_INTEGRATION
     result = gtkosx_application_get_executable_path();
     strncpy (path2, result, buf_size - 1);
     g_free (result);
     g_print ("Application Path %s\n", path2);
 #elif defined GNC_PLATFORM_OSX
-    /* Native Mac, but not Aqua */
-    size2 = buf_size;
-    if (_NSGetExecutablePath (path2, &size2) != 0)
     {
-        /* buffer not big enough or some other error */
-        if (error)
-            *error = GNC_GBR_INIT_ERROR_NOMEM;
-        return NULL;
+        /* Native Mac, but not Aqua */
+        uint32_t size2 = buf_size;
+        if (_NSGetExecutablePath (path2, &size2) != 0)
+        {
+            /* buffer not big enough or some other error */
+            if (error)
+                *error = GNC_GBR_INIT_ERROR_NOMEM;
+            return NULL;
+        }
```

Comment:
I now realise that the block is not needed anyway -- those `ifdef`s are confusing.

BTW I notice that the C compile options include -Wdeclarations-after-statement, although it is not mentioned in the coding standard. Would it be OK to remove it?

Reply:
> BTW I notice that the C compile options include -Wdeclarations-after-statement, although it is not mentioned in the coding standard. Would it be OK to remove it?

Yeah. I don't follow the C standard as closely as I do the C++ one so I didn't realize that C99 allows mixing decls and statements, I thought it was just a gcc extension. Since we require C11 (and should probably bump it to C17 so it matches the C++ standard requirement) we can certainly allow it. I'll go further and say we should encourage it in the coding standard because it makes it much easier to declare and initialize instead of assigning, and also allows clustering variables and their uses. @gjanssens Any comment on that?

Reply:
I'm all for allowing this. I agree it makes code much more readable.

Reply:
I'm fine with switching to C17 as well by the way, though I have no idea what improvements have happened since C98 we could make use of.

Reply:
I [dug into that](https://en.cppreference.com/w/c/language/history) a little bit yesterday. It turns out that there's only one significant change between C11 and C17, the behavior if you try to `realloc(foo, 0)`.  We don't use `realloc` so it doesn't matter to us, so there's no reason to change from C11.
 
Some C98 and C11 features that would let us make C code more C++-ish: Declaring variables at point of use, type `bool` instead of `TRUE` and `FALSE` macros to int, `inline` functions, and maybe some others that need a little more study to understand.

Reply:
OK with me, just need to remember to do it.